### ISSUE TITLE
implement basic medication module

### ISSUE THREAD

**Issue by jackcmeyer** [HospitalRun/hospitalrun-frontend#2229]
## 🚀 Feature Proposal

This issue is to create a very basic medication module. When this issue is finished, I would expect to be able to: 

- create a new medication request
- view a list of all medication requests
- view a single medication request

## Technical Notes
A medication request is defined as:

```
{
  id: string
  requestedBy: string // id of the user who created the request
  requestedOn: string // date when the medication was requested
  status: active | on hold | cancelled | completed | entered in error | stoped | draft | unknown
  intent: proposal | plan | order | original order | reflex order | filler order | instance order | option
  priority: routine | urgent | asap | stat
  medication: string
  patient: string // the id of the patient the request is for
  note: string
  quantity: { value: number, unit: string }
}
```

- There should be a text box which allows me to search patients
- id, requestedBy, requestedOn should be autogenerated

- The table should show: 
requested date, patient name, medication, quantity, status

- Add new set of permissions for each viewing and making medication requests
- Add sidebar link
- Add "Request Medication" to quick create button in navbar

**Comment by blestab** [comment]
Hi @jackcmeyer not sure what the timelines on this are, but would want to try give it a go. 

What is the expected ETA for this in terms of the project milestones?

**Comment by jackcmeyer** [comment]
@blestab no specific timeline :) however, there is a lot of work to be done here, so as soon as this issue gets done we can unblock a lot of that work. 

Feel free to reach out anytime you feel like you're stuck!

**Comment by blestab** [comment]
Thanks @jackcmeyer, i'll get right on it
